backend/src/main/java/outdooractivity/backend/security/SecurityConfig.java

```diff
@@ -0,0 +1,40 @@
+package outdooractivity.backend.security;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig {
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return Argon2PasswordEncoder.defaultsForSpringSecurity_v5_8();
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
+        requestHandler.setCsrfRequestAttributeName(null);
+        return http.csrf(csrf -> csrf.csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse()).csrfTokenRequestHandler(requestHandler))
+
+                .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.ALWAYS))
+                .httpBasic().and()
+                .authorizeHttpRequests()
+                .requestMatchers("/api/adventures").permitAll()
+                .requestMatchers("/api/add", "/api/update/**", "/api/adventures/**").authenticated()
```

Comment:
optional: for many routes its better to read when you create an array and pass the array to the matcher.
I think something like that should work:
`String[] protectedRoutes = {"/api/add", "/api/update/**", "/api/adventures/**"}`
and then :
`.requestMatchers(protectedRoutes).authenticated()`